server/lib/statvar_hierarchy_search.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Contains functions for building and using the search index for stat var
+# hierarchy based off all stat vars and stat var groups in our knowledge graph.
+# The search index should be built on Flask server initialization and saved in
+# the app config with the key "STAT_VAR_SEARCH_INDEX", and during search, it can
+# be accessed with current_app.config['STAT_VAR_SEARCH_INDEX'] to get the
+# results for each token in the search query.
+#
+# The index will be a dictionary of token to dictionary of ids (of stat vars and
```

Comment:
Nit: top level comments should be multi-line strings: https://google.github.io/styleguide/pyguide.html#382-modules

Reply:
done